site/ic_data_repo/symplectic_interface.py

```
@@ -0,0 +1,238 @@
+                        attrib=dict(scheme="orcid"),
+                    )
+                    orcid_element.text = identifier.get("identifier")
+
+        # Licence/Rights
+        rights = metadata.get("rights", [])
+        if rights:
+            licence_element = etree.SubElement(
+                native_element,
+                self.api_qname("field"),
+                attrib={
+                    "name": "c-licence",
+                    "type": "text",
+                    "display-name": "Licence",
+                },
+            )
+            licence_text_element = etree.SubElement(
+                licence_element, self.api_qname("text")
+            )
+            licence_text_element.text = rights[0].get("id")
+
+        doi = None
+        for identifier in metadata.get("identifiers", []):
+            if identifier.get("scheme", "").lower() == "doi":
+                doi = identifier.get("identifier")
+                break
+        if doi:
+            self.add_doi_subtree(native_element, "c-validated-doi", "DOI")
+
+        version = metadata.get("version")
+        if not version:
+            version = record.get("custom_fields", {}).get("imperial:dart_id")
+        if version:
+            version_element = etree.SubElement(
+                native_element,
+                self.api_qname("field"),
+                attrib={
+                    "name": "version",
+                    "type": "text",
+                    "display-name": "Version",
+                },
+            )
+            version_text_element = etree.SubElement(
+                version_element, self.api_qname("text")
+            )
+            version_text_element.text = str(version)
+
+        pub_date = metadata.get("publication_date")
+        if pub_date:
+            try:
+                pub_date_obj = datetime.fromisoformat(pub_date)
+            except Exception:
+                pub_date_obj = datetime.now()
```

The publication date will always be present and in a known format (as in the shared metadata examples). We don't want the exception handling here as any failure to convert the date should trigger human investigation rather than silently passing incorrect information.

---

My mistake I was having some trouble with the single dates when they only provide a year so i just used this to make it pass will fix it now